### ISSUE TITLE
Documentation examples broken — nn.Sequential / AdamW API mismatch

### ISSUE THREAD

**Issue by grillcheese123** [Grillcheese-AI/grilly#37]
## Bug

Running the following example from the documentation fails:

```python
import numpy as np
from grilly import nn
from grilly.optim import AdamW

model = nn.Sequential(
    nn.Linear(784, 256),
    nn.ReLU(),
    nn.Linear(256, 10),
)

optimizer = AdamW(model.parameters(), lr=1e-3)
loss_fn = nn.CrossEntropyLoss()

x = np.random.randn(32, 784).astype(np.float32)
targets = np.random.randint(0, 10, (32,))

logits = model(x)
loss = loss_fn(logits, targets)
grad = loss_fn.backward(np.ones_like(loss), logits, targets)

model.zero_grad()
model.backward(grad)
optimizer.step()
```

The documentation examples need to be updated to match the current API.

## Expected

All README / docs examples should run without errors on `grilly >= 1.0.0`.

## Action

- Audit all examples in README.md and docs/
- Update to match current API signatures
- Add a CI smoke test that runs the doc examples

**Comment by grillcheese123** [comment]
## Working example

The two issues are:
1. `loss_fn.backward()` signature is `(grad_output, input, target)` — pass the loss itself, not `np.ones_like(loss)`
2. `model.backward()` requires the input: `model.backward(grad, x)`

Here's the corrected version that runs and trains:

```python
import numpy as np
from grilly import nn
from grilly.optim import AdamW

# Build model
model = nn.Sequential(
    nn.Linear(784, 256),
    nn.ReLU(),
    nn.Linear(256, 10),
)

optimizer = AdamW(model.parameters(), lr=1e-3)
loss_fn = nn.CrossEntropyLoss()

# Data
x = np.random.randn(32, 784).astype(np.float32)
targets = np.random.randint(0, 10, (32,))

# Forward
logits = model(x)

# Loss
loss = loss_fn(logits, targets)
print(f"loss: {loss.mean():.4f}")

# Backward — pass (loss, logits, targets) and model.backward needs input x
grad = loss_fn.backward(loss, logits, targets)
model.zero_grad()
model.backward(grad, x)
optimizer.step()

# Verify improvement
logits2 = model(x)
loss2 = loss_fn(logits2, targets)
print(f"loss after step: {loss2.mean():.4f} (improved: {loss2.mean() < loss.mean()})")
```

Output:
```
loss: 3.4069
loss after step: 1.6958 (improved: True)
```
